src/Clock.tsx

```diff
@@ -0,0 +1,57 @@
+import { Component } from 'react';
+
+interface State {
+  today: Date,
+}
+
+interface Prop {
+  clockName: string;
+}
+
+export class Clock extends Component<Prop, State> {
+  state = {
+    today: new Date(),
+  };
+
+  timerId = Number();
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ today: new Date() });
+      // eslint-disable-next-line
+      console.info(this.state.today.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Prop) {
+    const { clockName } = this.props;
+    const { clockName: prevClockName } = prevProps;
+
+    if (clockName !== prevClockName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevClockName} to ${clockName}`);
+    }
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
+
+  render() {
+    const { clockName } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {this.state.today.toUTCString().slice(-12, -4)}
```

Comment:
this part of code is repeated. A very important programming principle is DRY - Don't Repeat Yourself. Think of a way to separate this logic